src/audio/volume/Kconfig.simd

```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+comment "volume optimization level select"
+
+choice "VOLUME_SIMD_LEVEL_SELECT"
+	prompt "choose which SIMD level used for volume module"
+
+	config VOLUME_SIMD_TOOLCHAIN_SELECT
+		prompt "SIMD will selected by toolchain pre-defined header"
+		bool
+		help
+			When this was selected, optimization level will be determined
+			by toolchain pre-defined macros in core isa header file.
+
+	config VOLUME_HIFI3
+		prompt "choose HIFI3 intrinsic optimized volume module"
+		bool
+		help
+			This option used to build HIFI3 intrinsic optimized volume code
+
+	config VOLUME_HIFI4
+		prompt "choose HIFI4 intrinsic optimized volume module"
+		bool
```

Comment:
I still think we should display the choices depend on the specific compiled target devices, for example, we should not show the VOLUME_HIFI4 if compile a tgl binary.

Reply:
Yes this would be better but it requires duplicating all the `core-isa.h` files in Kconfig TWICE: once in Zephyr and another time in XTOS. This has been just discussed above, in my prototype https://github.com/zephyrproject-rtos/zephyr/pull/67513 + #8733 and elsewhere.

For now let's just provide the per-component selection (that most users should not change) and revisit later.

Reply:
It would be a fairly minor convenience: very few people are expected to change these and the few people who will use these are expected to know what is the default that they are manually overriding.